File: instrumentation/vertx-core-3.6.0/src/main/java/com/nr/vertx/instrumentation/InboundWrapper.java
/*
 *
 *  * Copyright 2020 New Relic Corporation. All rights reserved.
 *  * SPDX-License-Identifier: Apache-2.0
 *
 */

package com.nr.vertx.instrumentation;

import com.newrelic.api.agent.ExtendedInboundHeaders;
import com.newrelic.api.agent.HeaderType;
import io.vertx.core.http.HttpClientResponse;

import java.util.List;

public class InboundWrapper extends ExtendedInboundHeaders {

    private final HttpClientResponse response;

    public InboundWrapper(HttpClientResponse response) {
        this.response = response;
    }

    @Override
    public HeaderType getHeaderType() {
        return HeaderType.HTTP;
    }

    @Override
    public String getHeader(String name) {
        return response.getHeader(name);
    }

    @Override
    public List<String> getHeaders(String name) {
        return response.headers().getAll(name);
    }

}

File: instrumentation/apache-pekko-http-3_1/src/main/scala/com/agent/instrumentation/org.apache.pekko/http/InboundWrapper.scala
/*
 *
 *  * Copyright 2024 New Relic Corporation. All rights reserved.
 *  * SPDX-License-Identifier: Apache-2.0
 *
 */

package com.agent.instrumentation.org.apache.pekko.http

import org.apache.pekko.http.scaladsl.model.HttpRequest
import com.newrelic.api.agent.{ExtendedInboundHeaders, HeaderType}

import java.util
import scala.jdk.javaapi.CollectionConverters

class InboundWrapper(request: HttpRequest) extends ExtendedInboundHeaders {

  def getHeaderType: HeaderType = {
    HeaderType.HTTP
  }

  def getHeader(name: String): String = {
    request.headers.find(header => header.is(name.toLowerCase)).map(header => header.value).orNull
  }

  override def getHeaders(name: String): util.List[String] = {
    val headers = request.headers.filter(header => header.is(name.toLowerCase)).map(header => header.value)
    if (headers.isEmpty) {
      return null
    }
    CollectionConverters.asJava(headers)
  }

}

